model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -457,10 +457,9 @@ message ParDoPayload {
   // be placed in the pipeline requirements.
   map<string, TimerFamilySpec> timer_family_specs = 9;
 
-  // Whether the DoFn is splittable
-  bool splittable = 6;
-
-  // (Required if splittable == true) Id of the restriction coder.
+  // (Optional) Only set when this ParDo contains a splittable DoFn.
```

Comment:
I'm not convinced that using the presence of a restriction coder is better than explicitly marking things as splittable, but could possibly be persuaded.

Reply:
`splittable` is invalid without the other field being set so why have it?
It would allow for bugs where one is set and the other is not.

Reply:
Because it's not as obvious. But I can see the pros and cons.